projects/igniteui-angular/src/lib/combo/combo.component.ts

```diff
@@ -183,6 +192,37 @@ export class IgxComboComponent extends DisplayDensityBase implements IgxComboBas
     @Input()
     public overlaySettings: OverlaySettings = null;
 
+    /**
+     * Specifies how the combo component handles selection.
+     *
+     * When set to single selection, the combo will allow only one of its items to be selected.
+     * When a new one is chosen, previous selection is erased.
+     *
+     * Multiple selection allows multiple items to be chosen.
+     *
+     * @default 'multiple'
+     *
+     * @example
+     * ```html
+     * <igx-combo [data]="comboData" selection="single" [(ngModel)]="selected">
+     * </igx-combo>
+     * ```
+     */
+    @Input()
+    public get selectionMode(): ComboSelectionMode {
+        return this._selectionMode;
+    }
+
+    public set selectionMode(val: ComboSelectionMode) {
+        if (this._selectionMode !== val) {
+            this._selectionMode = val;
+            const selection = this.selectedItems();
+            if (this._selectionMode === ComboSelectionMode.single && selection.length > 1) {
+                this.setSelection(new Set(this.validateSelectionSize(selection)));
```

Comment:
No need to validate selection size. `setSelection` will do this for you.
```suggestion
                this.setSelection(new Set(selection));
```